### ISSUE TITLE
[Suggestion] Advanced Macros

### ISSUE THREAD

**Issue by Cypphi** [MeteorDevelopment/meteor-client#1198]
**Describe the feature**
Add more options to macros.
These options would be:

- Delay between messages
- A delay for how long it should wait before sending the message
- How many times to loop a specific message
**All these options should be configurable for every singe message, this would be useful for sending specific messages in the right order instead of all at once.**

**Additional info**
secks :trollface:

**Comment by Cypphi** [comment]
Edit, maybe make macros toggleable as well so you can loop the same commands with a specific interval forever.

**Comment by RedCarlos26** [comment]
I agree

**Comment by BobIsMyManager** [comment]
> Edit, maybe make macros toggleable as well so you can loop the same commands with a specific interval forever.

You can just edit your message 😐

**Comment by D3V1LC0D3R** [comment]
it would be ice if the macros could be upgraded to something more general like a scripting language with functions, ifs & elses, ...

**Comment by Cypphi** [comment]
> it would be nice if the macros could be upgraded to something more general like a scripting language with functions, ifs & elses, ...

Agreed, it would be cool to have a scripting API which makes usage of something like JS similar to how Astolfo client does it

**Comment by BobIsMyManager** [comment]
femboy client

**Comment by Cypphi** [comment]
Looking back at this, if this wouldn't be implemented, at least add onEnable and onDisable events to macros 🙏🏻

**Comment by DrParanoya** [comment]
> it would be nice if the macros could be upgraded to something more general like a scripting language with functions, ifs & elses, ...

Starscript: am I a joke to you?
